apis/policies/v1alpha2/admissionpolicy_webhook.go

```diff
@@ -0,0 +1,83 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// log is for logging in this package.
+var admissionpolicylog = logf.Log.WithName("admissionpolicy-resource")
+
+func (r *AdmissionPolicy) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	err := ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+	if err != nil {
+		return fmt.Errorf("failed enrolling webhook with manager: %w", err)
+	}
+	return nil
+}
+
+//+kubebuilder:webhook:path=/mutate-policies-kubewarden-io-v1alpha2-admissionpolicy,mutating=true,failurePolicy=fail,sideEffects=None,groups=policies.kubewarden.io,resources=admissionpolicies,verbs=create;update,versions=v1alpha2,name=madmissionpolicy.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Defaulter = &AdmissionPolicy{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *AdmissionPolicy) Default() {
+	admissionpolicylog.Info("default", "name", r.Name)
+	if r.ObjectMeta.DeletionTimestamp == nil {
+		controllerutil.AddFinalizer(r, constants.KubewardenFinalizer)
+	}
+}
+
+//+kubebuilder:webhook:path=/validate-policies-kubewarden-io-v1alpha2-admissionpolicy,mutating=false,failurePolicy=fail,sideEffects=None,groups=policies.kubewarden.io,resources=admissionpolicies,verbs=create;update,versions=v1alpha2,name=vadmissionpolicy.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &AdmissionPolicy{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *AdmissionPolicy) ValidateCreate() error {
```

Comment:
what do you have in mind for this? why would you block an AdmissionPolicy to target a specific PolicyServer?

Reply:
Policy servers are cluster wide resources. I could want to create 3 policy servers, one per tenant for example.

This is a problem with clusteradmissionpolicies too, but we need to tackle this for admissionpolicies as well. It's more visible because I as a restricted user able to create admission policies on my namespace can target my policy to be executed in `policy-server-a`, `policy-server-b` and `policy-server-c`, but I am `tenant-a` and would only be supposed to target `policy-server-a`, for example.